rest-service/manager_rest/resource_manager.py

```diff
@@ -2313,21 +2315,28 @@ def _verify_dependencies_not_affected(self,
             )
         )
 
-    def _excluded_component_creator_id(self, deployment):
-        component_creator_deployment = \
-            [x.source_deployment.id for x in
-             deployment.target_of_dependency_in if
-             'component' in x.dependency_creator.split('.')[0]]
-        if component_creator_deployment:
+    def _excluded_component_creator_ids(self, deployment, dep_graph):
+        dep_graph.create_dependencies_graph()
+        component_creator_deployments =\
```

Comment:
```suggestion
    component_creator_deployments = []
    for d in dep_graph.retrieve_dependent_deployments(deployment.id):
        if d['dependency_type'] == 'component':
            component_creator_deployments.append(d['deployment'])
```

The same number of lines, and a lot more readable.